code/main/consent/consentState.brs

```diff
@@ -0,0 +1,139 @@
+' ********************** Copyright 2024 Adobe. All rights reserved. **********************
+' *
+' * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+' * you may not use this file except in compliance with the License. You may obtain a copy
+' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+' *
+' * Unless required by applicable law or agreed to in writing, software distributed under
+' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+' * OF ANY KIND, either express or implied. See the License for the specific language
+' * governing permissions and limitations under the License.
+' *
+' *****************************************************************************************
+
+' ************************************ MODULE: Consent ***************************************
+
+function _adb_isConsentStateModule(obj as object) as boolean
+    return (obj <> invalid and obj.type = "com.adobe.module.consentState")
+end function
+
+function _adb_ConsentState(configurationModule as object) as object
+    if not _adb_isConfigurationModule(configurationModule) then
+        _adb_logError("ConsentState::_adb_ConsentState() - configurationModule is not valid.")
+        return invalid
+    end if
+
+    consentState = _adb_AdobeObject("com.adobe.module.consentState")
+    consentState.Append({
+        _configurationModule: invalid,
+        _collectConsent: invalid,
+
+        _init: sub(configurationModule as object)
+            _adb_logVerbose("ConsentState::_init() - Initializing consent state.")
+            m._configurationModule = configurationModule
+            m.getCollectConsent()
+        end sub,
+
+        setCollectConsent: sub(collectConsent as dynamic)
```

Comment:
Why only allow setting and retrieving of the collect consent and not a map of all consents?

Reply:
Yes, we are not saving other consents, or allowing customers to append to already present consents. And SDK only reads and honors collect consent at this point and that too we read from Konductor response.

Reply:
So, for Roku, you are saying customers will need to pass in all of their Consents each time they call setConsent(), except for "collect" because there is a configured default? Why are we making this change? Is it because of space limitations in Roku for storing persisted data?